site/admin/theatres/create.php

```diff
@@ -0,0 +1,107 @@
+<?php
+include_once('init.php');
+PrintHeader('Create Theatre', 'Create Theatre');
+
+
+if(isset($_REQUEST['TheatreFormSubmitted'])){
+	//die("the form was submitted");
+	if(!@$_REQUEST['TheatreName']){
+		$FormErrors['TheatreName'] = "Theatre Name required";
+	}
+	if(!@$_REQUEST['TheatreDescription']){
+		$FormErrors['TheatreDescription'] = "Theatre must have a description";
+	}
+	if(!@$_REQUEST['Email'] || strpos($_REQUEST['Email'], '@') == false){
+		$FormErrors['Email'] = "Must be valid email address";
+	}
+	if(!@$_REQUEST['StreetAddress']){
+		$FormErrors['StreetAddress'] = "Address required";
+	}
+	if(!@$_REQUEST['City']){
+		$FormErrors['City'] = "City required";
+	}
+	if(!@$_REQUEST['State']){
+		$FormErrors['State'] = "State required";
+	}
+	if(!@$_REQUEST['Zip']){
+		$FormErrors['Zip'] = "Zip required";
+	}
+	if(!@$_REQUEST['Website'] || strpos($_REQUEST['Website'], '.') == false){
+		$FormErrors['Website'] = "Valid website required";
+	}
+	if(!@$_REQUEST['DateFormed']){
+		$FormErrors['DateFormed'] = "Date required";
+	}
+	if(!@$_REQUEST['Type']){
+		$FormErrors['Type'] = "Type required";
+	}
+	if(sizeof($FormErrors) == 0){
+		SaveForm($_REQUEST['TheatreName'], $_REQUEST['TheatreDescription'],
```

Comment:
Note: I realize you're just doing what I showed you in the forms lesson. But here are some things to improve when doing this in real life:

As your code base grows, you'll start to find that the names you choose for variables and functions becomes more and more important. The name `SaveForm` isn't descriptive enough because you're going to end up with a lot of forms, and you can't have two functions with the same name.

Additionally, you want your PHP functions to be atomic, and agnostic about how they're used. What I mean is that this function doesn't "save a form". This function "creates a theatre". Sometimes you might want to create theatres via some other method, like syncing with a third-party database or via a CSV import. So I'd call this function `InsertTheatre` or `CreateTheatre`